zipkin/src/zipkin_reporter_impl.cc

```diff
@@ -21,7 +21,12 @@ void ReporterImpl::reportSpan(const Span &span) {
   bool is_full;
   {
     std::lock_guard<std::mutex> lock(write_mutex_);
-    num_spans_reported_ += spans_.addSpan(span);
+
+    if (spans_.addSpan(span)) {
+      num_spans_reported_++;
+    } else {
+      dropped_span_count_++;
```

Comment:
This might be able to have loser memory ordering, but that might be a premature optimization.